api/v1alpha1/rollout_types.go

```diff
@@ -111,6 +117,15 @@ type CanaryStep struct {
 	// Pause defines a pause stage for a rollout, manual or auto
 	// +optional
 	Pause RolloutPause `json:"pause,omitempty"`
+	// FailureThreshold indicates how many failed pods can be tolerated in the pods upgraded in current batch.
```

Comment:
plz comment explicitly in the code about the percent base, is the base the workload replicas or the current bat ch replicas